### PR TITLE
harmony-one 4.2.1 (new formula)

### PR DESCRIPTION
**Pull request by cylim**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/harmony-one.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
+class HarmonyOne < Formula
+  desc "CLI tool to interact with the Harmony Blockchain"
+  homepage "https://docs.harmony.one/"
+  url "https://github.com/harmony-one/harmony/archive/refs/tags/v4.2.1.tar.gz"
+  sha256 "d29609a99a3a2e9031d2a26414ffdc3e271c3bf9cf9331dfcb321715fef682d4"
+  license "LGPL-3.0-or-later"
+
+  depends_on "go@1.16" => :build
```
Comment (SMillerDev) [inline Formula/harmony-one.rb:8]: Why doesn't it work with the latest Go?

File: Formula/harmony-one.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  depends_on arch: :x86_64
```
Comment (SMillerDev) [inline Formula/harmony-one.rb:10]: Why doesn't it work on new macs?

File: Formula/harmony-one.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    url "https://github.com/harmony-one/mcl.git",
+      revision: "99e9aa76e84415e753956c618cbc662b2f373df1"
+  end
+
+  def install
+    # See https://github.com/golang/go/issues/26487
+    ENV.O0 if OS.linux?
+
+    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULE"] = "auto"
```
Comment (SMillerDev) [inline Formula/harmony-one.rb:39]: This shouldn't be needed in modern software.

File: Formula/harmony-one.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      bin.install "dist/hmy"
+      prefix.install_metafiles
+    end
+  end
+
+  def caveats
+    <<~EOS
+      Invoke 'hmy cookbook' for examples of the most common, important usages
+      To gerenate docs in your current directories, run: 'hmy docs'
+    EOS
```
Comment (SMillerDev) [inline Formula/harmony-one.rb:72]: This should be in the upstream documentation, it's not homebrew specific.

Comment (cylim) [reply Formula/harmony-one.rb:72]: I will remove it in next force push. Thanks for letting me knows. 😄

File: Formula/harmony-one.rb (hunk 3 above)
Comment (cylim) [reply Formula/harmony-one.rb:39]: GOPATH is needed due to it is using in the Makefile to reference to other resources.

File: Formula/harmony-one.rb (hunk 2 above)
Comment (cylim) [reply Formula/harmony-one.rb:10]: I will try to debug this after I solved the linux build issues. It was failing in the CI and I can't test it without M1 device. :)

File: Formula/harmony-one.rb (hunk 1 above)
Comment (cylim) [reply Formula/harmony-one.rb:8]: probably https://github.com/golang/go/issues/46645 issue. Will try to get the team to upgrade. 😄

File: Formula/harmony-one.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  depends_on "go@1.16" => :build
+
+  depends_on arch: :x86_64
+  depends_on "gmp"
+  depends_on "openssl@1.1"
+
+  on_linux do
+    depends_on "gcc"
+    depends_on "glibc"
```
Comment (dawidd6) [inline Formula/harmony-one.rb:16]: Why? Does this software really need latest glibc?

Comment (cylim) [reply Formula/harmony-one.rb:16]: The software doesn't need the latest. 

I just figured out that Homebrew for Linux required those dependencies too, I had removed it from my local repo, but haven't hard push yet. 

I am waiting for other fixes, then hard push at once. 😄

**Pull request closed without merging**